homeassistant/components/tankerkoenig/sensor.py

```diff
@@ -125,9 +135,15 @@ def device_state_attributes(self):
             ATTR_HOUSE_NUMBER: self._house_number,
             ATTR_POSTCODE: self._postcode,
             ATTR_CITY: self._city,
-            ATTR_LATITUDE: self._latitude,
-            ATTR_LONGITUDE: self._longitude,
         }
+
+        if self._show_on_map:
+            attrs[ATTR_LATITUDE] = self._latitude
+            attrs[ATTR_LONGITUDE] = self._longitude
+        else:
+            attrs["lat"] = self._latitude
+            attrs["long"] = self._longitude
```

Comment:
```suggestion
```
I have also now looked at the other integrations you proposed, and it seems there is no standard there. Some don't show anything, some show "lat" and "long", other "lati" and "long",...

I think your initial proposal (i.e. just not adding the attributes if show_on_map is false) is the best solution. I was hoping for an extra thing that would allow having latitude and longitude, but hide the entities from the map, but that seems to be the standard way of hiding them.
Sorry about making you change this back and forth :-(